plugins/woocommerce/tests/php/includes/wc-core-functions-test.php

```diff
@@ -224,4 +224,53 @@ function () {
 		$this->assertEquals( '', $result['state'] );
 		remove_all_filters( 'woocommerce_geolocate_ip' );
 	}
+
+	/**
+	 * Test wc_delete_transients() function.
+	 */
+	public function test_wc_delete_transients() {
+		// Set up test transients.
+		$transient_name1 = 'wc_test_transient_1';
+		$transient_name2 = 'wc_test_transient_2';
+
+		set_transient( $transient_name1, 'test_value_1', HOUR_IN_SECONDS );
+		set_transient( $transient_name2, 'test_value_2', HOUR_IN_SECONDS );
+
+		// Verify transients exist before deletion.
+		$this->assertEquals( 'test_value_1', get_transient( $transient_name1 ) );
+		$this->assertEquals( 'test_value_2', get_transient( $transient_name2 ) );
+
+		// Delete the transients.
+		wc_delete_transients( array( $transient_name1, $transient_name2 ) );
+
+		// Force WordPress to refresh its memory cache.
+		wp_cache_flush();
```

Comment:
Is this required? If so, I wonder if it should be part of `wc_delete_transients` itself.

Reply:
Yeah good point, this forced me to dive a bit deeper into how we delete options and they also handle cache deletion which I've now added and mirrored in this functions functionality https://github.com/woocommerce/woocommerce/pull/55931/commits/12626b373f095286206e243952bff575521142fb